### PR TITLE
Prepare for PHP 8.0

### PR DESCRIPTION
**Pull request by lifinsky**
- remove incompatible dev dependencies;
  - update PSR to v12;
  - update tests for Codeception 4;
  - add .gitignore.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,18 +11,17 @@
     ],
     "minimum-stability": "stable",
     "require": {
-        "php": "^7.2",
+        "php": "^7.2 | ^8.0",
```
Comment (mkorkmaz) [inline composer.json]: Maybe we should make it ^7.3|^8.0 since 7.2 support will end completely at end of Nov.

File: composer.json [hunk 2]
```diff
@@ -11,18 +11,17 @@
...
-        "codeception/codeception": "^2.5",
-        "php-coveralls/php-coveralls": "^2.1",
-        "squizlabs/php_codesniffer": "^3.3",
+        "codeception/codeception": "^4.1",
+        "squizlabs/php_codesniffer": "^3.5",
         "predis/predis": "^1.1",
-        "ext-redis": "*"
+        "ext-redis": "*",
+        "codeception/module-asserts": "^1.3",
+        "codeception/assert-throws": "^1.1"
```
Comment (mkorkmaz) [inline composer.json]: I think we dont neeed assert-throws. module asserts will be enough.

File: composer.json [hunk 3]
```diff
@@ -39,10 +38,8 @@
         }
     },
     "scripts": {
-        "unit-tests": "vendor/bin/codecept run unit --coverage",
-        "bc-check": "vendor/bin/roave-backward-compatibility-check",
```
Comment (mkorkmaz) [inline composer.json:43]: I understand this library prevents further upgrade, I am ok to remove it temporarily.

File: tests/ModuleTest.php [hunk 4]
```diff
@@ -61,15 +55,16 @@ public function shouldRunCommandOriginalPredisClientsCommandSuccessfully() : voi
...
-    public function shouldFailForMagicMethodCall() : void
+    public function shouldFailForMagicMethodCall(): void
     {
         $redisClient = new Predis\Client();
         /**
          * @var Module
          */
         $module = Module::createWithPredis($redisClient);
-        $module->invalidMethod('test');
+        $this->assertThrows(InvalidCommandException::class, function () use ($module) {
```
Comment (mkorkmaz) [inline tests/ModuleTest.php]: can we just use this assert instead of assertThrows:

`$this->expectException(InvalidCommandException::class);`

**Review by mkorkmaz — COMMENTED** [review]
@lifinsky Thanks Serg for your PR. Would you consider to change the points in my comments. After that I can merge the PR and then I can update travis configuration to tag the new version.

File: composer.json (hunk 2 above)
Comment (lifinsky) [reply composer.json]: Ok, my mistake. I always use phpunit, may be I forgot that API is the same.

**Comment by lifinsky** [comment]
Done

**Pull request merged**